### ISSUE TITLE
deviceType clkg is not showing in homekit

### ISSUE THREAD

**Issue by theadamzaft** [tuya/tuya-homebridge#166]
Integrated tuya plugin into homebridge.
Added account and linked to IOT
in the logs i can see that connection is ok, and the plugin gets my device info, but nothing is displayed in accessories.
deviceType : clkg

**Comment by adam-is-here** [comment]
any news?

**Comment by talrhv** [comment]
Hi, I have the same issue so I debugged it myself and added to the category switch case 'clkg'
now I can see the curtain switch in my home app but the problem is now that when to window is closed the home app shows "opened" and when it is open it shows "closed". please check this issue. thanks!

**Comment by talrhv** [comment]
> any news?

Hi Again,
I added support for the curtain switch and fixed the problem that I described on the previous reply.
I'm not sure how to share it with the tuya developers so they can add it to the plugin.

**Comment by adam-is-here** [comment]
> > any news?
> 
> Hi Again, I added support for the curtain switch and fixed the problem that I described on the previous reply. I'm not sure how to share it with the tuya developers so they can add it to the plugin.

Did you tried to open a pull request?

**Comment by talrhv** [comment]
> > > any news?
> > 
> > 
> > Hi Again, I added support for the curtain switch and fixed the problem that I described on the previous reply. I'm not sure how to share it with the tuya developers so they can add it to the plugin.
> 
> Did you tried to open a pull request?

I'm not sure how. a friend will help me tomorrow.
for now you can download my fork.

**Comment by talrhv** [comment]
> > > any news?
> > 
> > 
> > Hi Again, I added support for the curtain switch and fixed the problem that I described on the previous reply. I'm not sure how to share it with the tuya developers so they can add it to the plugin.
> 
> Did you tried to open a pull request?

I made a pull request

**Comment by adam-is-here** [comment]
Thanks!

**Comment by DanAykroyd256** [comment]
Hi @talrhv,

I'm having [a similar problem](https://github.com/tuya/tuya-homebridge/issues/208) with a breaker showing in the Homebridge log as being detected, and even working from the Debug tab in the Tuya IoT Platform, but it doesn't show under the Accesories tab in Homebridge and I can't use it.

Would it be possible for you to let me know how exactly did you fix yours and if I could do the same for mine?

I truly appreciate it; thanks in advance.

**Comment by talrhv** [comment]
Hi,

send me the category name.

**Comment by DanAykroyd256** [comment]
Thanks @talrhv for your fast reply!

The category name is dlq; you can check all the details in my open issue [here](https://github.com/tuya/tuya-homebridge/issues/208). Please let me know if you need anything else from my side.

Best.

**Comment by talrhv** [comment]
does Homekit supports this kind of device?

**Comment by DanAykroyd256** [comment]
Yes, it should show up as a switch and the only attribute that should be there to control would be turning it on and off with the supported command I paster in my post, that gets picked up by this plug in in the logs (but doesn't show up in Accesories).

Device functions
{
"code":"switch_prepayment",
"desc":"switch prepayment",
"name":"switch prepayment",
"type":"Boolean",
"values":"{}"
},
{
"code":"energy_reset",
"desc":"energy reset",
"name":"energy reset",
"type":"Enum",
"values":"{"range":["empty"]}"
},
{
"code":"charge_energy",
"desc":"charge energy",
"name":"charge energy",
"type":"Integer",
"values":"{"unit":"kW·h","min":0,"max":999999,"scale":2,"step":1}"
},
**{
"code":"switch",
"desc":"switch",
"name":"switch",
"type":"Boolean",
"values":"{}"
}**

**Comment by talrhv** [comment]
I  think that I can present it as a switch on the home app but all the
other settings will be set through tuya’s app.

**Comment by DanAykroyd256** [comment]
Yes @talrhv, that's exactly how it should be, with only the switch being available and the rest through the HomeKit app.

Are you going to include it in your fork or will you let me know how to do it myself?

Thanks again for your help.

**Comment by Vasco203** [comment]
Hello,

I have exactly the same issue. How can it be fixed?
I see the clkg in the IoT page, in the Tuya app and it works both. It also shows up in the log in Hoobs but doesn't shows up in the accessories page.

Any help would be much appreciated.
Thanks!

Vasco

**Comment by DanAykroyd256** [comment]
Hi @Vasco203, could you please comment your situation on my [open issue](https://github.com/tuya/tuya-homebridge/issues/208) for our device type, so the developers of the plugin could see it?

Thanks.

**Comment by talrhv** [comment]
Download my fork instead of the main plugin.
Uninstall yours (keep confit file) and search on the plugins tab for
“talrhv” your will find only one result. Install it and it should work.

**Comment by DanAykroyd256** [comment]
Thanks @talrhv.

I've tried your plugin and the same still happens. The accesory gets listed in the log during start up, but it doesn't show up in the “Accesories” tab. Have you added the **dlq** category to it?

(I get this in the log, with all the parameters and actions supported listed, too)
Adding: Breaker  (dlq / xxxxxxxxxxxxxxx)

**Comment by talrhv** [comment]
no not yet. I didn't have the time to do it yet.

**Comment by talrhv** [comment]
my fork supports 'clkg' curtain switch. I'll try to add the dlq support on
the weekend.

**Comment by DanAykroyd256** [comment]
Thanks @talrhv, looking forward to this.

**Comment by DanAykroyd256** [comment]
Thanks @talrhv, it works great; it even “detected” and reflects the status of the switch if I manually switch it from the device itself, which was impossible to achieve with IFTTT or using Scenes as a workaround. You should really get in touch with the “official” plugin developers to add these devices you kept adding for us!

If I stay in your fork, how do updates to the official plugin reflect there? Do you keep updating it for every update to this plugin, or should I manually do something to update it with the “official” improvements (while maintaining your added devices!).

Thanks again man, I really appreciate this!

**Comment by DanAykroyd256** [comment]
Hi again @talrhv; here bothering you again :)

I've noticed another switch added to the same accesory named “switch_prepayment”. Would it be possible to only support the switch command, as you mentioned before, in order to not display that one there? I think that enabling prepayment changes the breaker to work only until a certain amount of energy has been consumed (and then shuts off), so it can create quite a mess if wrongly mispressed.

![image](https://user-images.githubusercontent.com/64184474/151681824-c22fce1d-29fd-4f50-8eb8-3b4a49b0c3a1.png)

Thanks!

**Comment by talrhv** [comment]
Hi, I’m happy that it works. I’ll keep updating it.

**Comment by DanAykroyd256** [comment]
Thanks again @talrhv! Would it be possible to only keep the “switch” action as an accesory and ignore/remove the “switch_prepayment” one?

**Comment by DanAykroyd256** [comment]
Hi again @talrhv. I've been trying to use the “ignore specific devices by their device id” section in your plugin, by adding a device ID to be ignore (and not show up in Homebridge), but it doesn't do anything, as the accesory continues to appear.

Is that functionality working and doing anything?

Thanks.

**Comment by DanAykroyd256** [comment]
Once agaiin @talrhv, sorry to bother you.

After further testing, I'm still not able to use your implementation because something weird is going one, related to what I mentioned before to that “switch_prepayment” action.

After configuring your plugin and adding the devices to HomeKit, each of this breakers gets imported as a group of two accesories (the “switch” action and the “switch_prepayment) action. After ungrouping them in HomeKit, setting completely different names for all of them, when I say (for example) ‘Hey Siri, turn on the AC’ (name of one of the breaker's switch action I renamed), Siri always acts on the “switch_prepayment” action and there is no way to make her work on the actual name I'm saying! It's like the devices are linked somehow at an even lower-level and Siri takes one of them to respond.

Take a look at this crazy scenario:

- Hey Siri, turn on AC <AC name of the switch accesory>
- Sure, I have turned on TRASH DEVICE DO NOT USE <actual name I gave to the switch_prepayment accesory>

She always responds for the switch_prepayment device… no matter what.

So, I'm back again with my request. Would it be possible for you to only allow the recognition of the “switch” action and ignore the “switch_prepayment” one?

Thanks for a millionth time, I'm just so close to making this work that I'm pretty much in tears :(

**Comment by lars05** [comment]
Hi @talrhv , 

appreciate your effort. With your fork I can now see my 3 CLKG curtain switches in the Home app. 
However, nothing happens when controling them through Home. The console also does not show any response when using the Home app

**Comment by Vasco203** [comment]
> Download my fork instead of the main plugin. Uninstall yours (keep confit file) and search on the plugins tab for “talrhv” your will find only one result. Install it and it should work.
> […](#)

Thanks, I would love to use it. Sorry for being a noob, but how can I install it? I'm using Hoobs and only have experience with installing plugins via the UI plugin directory so far. A little step by step guide would be much appreciated!
I already have the Tuya API set up etc at the IoT page of Tuya. So far so good.

TIA!

**Comment by talrhv** [comment]
You may need to remove the device from Homebridge cache. Go to homebridge
settings, you should find it there.

**Comment by talrhv** [comment]
I don’t think that I can help you without having the device itself for
debugging purposes.

**Comment by lars05** [comment]
> Hi @talrhv ,
> 
> appreciate your effort. With your fork I can now see my 3 CLKG curtain switches in the Home app. However, nothing happens when controling them through Home. The console also does not show any response when using the Home app

Any idea what I could do @talrhv ? 
See the log for referance. However, nothing happens when executed a close/open command in Home app. 

`TuyaOpenAPI response: {"result":{"access_token":"e911b63ac8ea59e0604a69077d","expire_time":7200,"platform_url":"https://openapi.tuyaeu.com","refresh_token":"d0ab0a6fb5ea16fa85cf3fa1a56","uid":"eu162437086Md4MD"},"success":true,"t":1644050082016} path = /v1.0/iot-01/associated-users/actions/authorized-login
TuyaOpenAPI request: method = get, endpoint = https://openapi.tuyaeu.com, path = /v1.0/iot-01/associated-users/devices, params = {"size":100}, body = null, headers = {"t":"1644050082045","client_id":"kaacegr9bbbgyxvzu","nonce":"760ec510-865e-11ec-bc39-4570eea8ad7f","Signature-Headers":"client_id","sign":"1DD6E66D1F0417E548DA9A00D55BC07554C7526EDF4CFECF9F73E228868","sign_method":"HMAC-SHA256","access_token":"e911b63ac8ea1fb0604a69077d","lang":"en","dev_lang":"javascript","dev_channel":"homebridge","devVersion":"1.5.0"}
TuyaOpenAPI response: {"result":{"devices":[{"active_time":1643892732,"biz_type":0,"category":"clkg","create_time":1624962467,"icon":"smart/icon/ay1496289297888GkNKB/4d93d2723eff19b097faf1702612e760.png","id":"632358008caab5e42727","ip":"87.133.129.232","lat":"53.5879","local_key":"2edb23fe5c9","lon":"10.0020","name":"Wohnzimmer rechts","online":true,"owner_id":"50233445","product_id":"237mvcniwsrjgqhl","product_name":"Smart Curtain Switch","status":[{"code":"control","value":"stop"}],"sub":false,"time_zone":"+01:00","uid":"eu16437086Md4MD","update_time":1643919447,"uuid":"632358008caab2727"},{"active_time":1643894793,"biz_type":0,"category":"clkg","create_time":1624970422,"icon":"smart/icon/ay1496289297888GkNKB/4d93d2723eff19b097faf1702612e760.png","id":"73318caab5e52f87","ip":"87.133.129.232","lat":"53.5879","local_key":"c7867e5e39ab","lon":"10.0020","name":"Wohnzimmer links","online":true,"owner_id":"50233445","product_id":"237mvcniwsrjgqhl","product_name":"Smart Curtain Switch","status":[{"code":"control","value":"stop"}],"sub":false,"time_zone":"+01:00","uid":"eu162437086Md4MD","update_time":1643919446,"uuid":"7331236185e52f87"},{"active_time":1643894874,"biz_type":0,"category":"clkg","create_time":1621106027,"icon":"smart/icon/ay1496289297888GkNKB/4d93d2723eff19b097faf1702612e760.png","id":"2506f4cfa2f94d95","ip":"87.133.129.232","lat":"53.5879","local_key":"af46b5b76502","lon":"10.0020","model":"WF-CS01","name":"Schlafzimmer","online":true,"owner_id":"50233445","product_id":"237mvcniwsrjgqhl","product_name":"Smart Curtain Switch","status":[{"code":"control","value":"stop"}],"sub":false,"time_zone":"+01:00","uid":"eu892437086Md4MD","update_time":1643919441,"uuid":"25006406f4f94d95"}],"has_more":false,"last_row_key":"37846BDAA1F3E45E69F191EC39E86EBA","total":3},"success":true,"t":1644050082229} path = /v1.0/iot-01/associated-users/devices
TuyaOpenAPI request: method = get, endpoint = https://openapi.tuyaeu.com, path = /v1.0/devices/functions, params = {"device_ids":"632358008caab5e4233123618caab5e52f87,25406f4cfa2f94d95"}, body = null, headers = {"t":"1644050082248","client_id":"kaacbbbgyxvzrhgu","nonce":"760ec510-865e-11ec-bc39-4570eea8ad7f","Signature-Headers":"client_id","sign":"7CF1D86A8AEECCA7BC189B3463C15460471B3BFA9A1DD09DAF08BDC86FE77030","sign_method":"HMAC-SHA256","access_token":"e911b63ac88c51fb0604a69077d","lang":"en","dev_lang":"javascript","dev_channel":"homebridge","devVersion":"1.5.0"}
TuyaOpenAPI response: {"result":[{"category":"clkg","devices":["632358caab5e42727","733123618c5e52f87","25006406fa2f94d95"],"functions":[{"code":"control","desc":"control","name":"control","type":"Enum","values":"{\"range\":[\"open\",\"stop\",\"close\"]}"}],"product_id":"237mvcniwsrjgqhl"}],"success":true,"t":1644050082414} path = /v1.0/devices/functions
Adding: Wohnzimmer rechts (clkg / 632358aab5e42727)
Creating New Accessory 632358008caab5e42727
Register Platform Accessory Wohnzimmer rechts
Adding: Wohnzimmer links (clkg / 733123aab5e52f87)
Creating New Accessory 733123618caab5e52f87
Register Platform Accessory Wohnzimmer links
Adding: Schlafzimmer (clkg / 250064cfa2f94d95)
Creating New Accessory 2500640cfa2f94d95
Register Platform Accessory Schlafzimmer
TuyaOpenAPI request: method = post, endpoint = https://openapi.tuyaeu.com, path = /v1.0/iot-03/open-hub/access-config, params = null, body = {"uid":"eu1643892437086Md4MD","link_id":"7622c240-865e-11ec-bc39-4570eea8ad7f","link_type":"mqtt","topics":"device","msg_encrypted_version":"1.0"}, headers = {"t":"1644050082454","client_id":"kaacegr9bbbgyxvzrhgu","nonce":"760ec510-865e-11ec-bc39-4570eea8ad7f","Signature-Headers":"client_id","sign":"A5375E432B2C21BB66156A9F5DE2F8AD3D7B3394D621D1D9A2626FF7D","sign_method":"HMAC-SHA256","access_token":"e911b63ac8ea51fb0604a69077d","lang":"en","dev_lang":"javascript","dev_channel":"homebridge","devVersion":"1.5.0"}
TuyaOpenAPI response: {"result":{"client_id":"cloud_65dcb26ccbef3148d2acb6415099","expire_time":7200,"password":"3dd462416bd2302188f6157","sink_topic":{"device":"cloud/token/out/{device_id}"},"source_topic":{"device":"cloud/token/in/0f621afdfd00e7911dc231b99c"},"url":"ssl://m1.tuyaeu.com:8883","username":"cloud_0f621afd36189efd00e7911dc231b99c"},"success":true,"t":1644050082609} path = /v1.0/iot-03/open-hub/access-config
TuyaOpenMQ connecting: ssl://m1.tuyaeu.com:8883
TuyaOpenMQ connected`

**Comment by DanAykroyd256** [comment]
HI @talrhv,

Is there anything that can be done to filter out the “switch_prepayment” action and leave only the “switch” one for device type “dlq”?

Thanks.

**Comment by talrhv** [comment]
you probably have the one without the percentage control. I will support it
in my next version which will probably publish in about a week or so.

**Comment by lars05** [comment]
> 

Great to hear. Let me know if you need any more details about the hardware.

**Comment by DanAykroyd256** [comment]
Hi @talrhv, are you still working on your fork?

Thanks.

**Comment by lars05** [comment]
I think he is no longer developing the fork. 
Has somebody already found another fork which works with clkg devices?

**Comment by talrhv** [comment]
Hi guys, I’m a little busy so I don’t have the time to maintain it. When
I’ll have the time I’ll read your emails and get back to you.
